lib/src/io/static/static_handler.dart

```
@@ -1,327 +1,445 @@
-      throw ArgumentError('defaultDocument must be a file name.');
-    }
-  }
-
-  final mimeResolver = contentTypeResolver ?? _defaultMimeTypeResolver;
+  return (final NewContext ctx) async {
+    final requestPath = ctx.remainingPath.path;
+    final filePath = p.join(resolvedRootPath, requestPath.substring(1));
 
-  return respondWith((final Request request) {
-    final segs = [fileSystemPath, ...request.url.pathSegments];
+    // Ensure file exists and is not a directory
+    final entityType = FileSystemEntity.typeSync(filePath, followLinks: false);
+    if (entityType == FileSystemEntityType.notFound ||
+        entityType == FileSystemEntityType.directory) {
+      return defaultHandler!(ctx);
+    }
 
-    final fsPath = p.joinAll(segs);
+    // Security check for symbolic links: ensure resolved path stays within root directory
+    var file = File(filePath);
+    final resolvedFilePath = file.resolveSymbolicLinksSync();
+    if (!p.isWithin(resolvedRootPath, resolvedFilePath)) {
+      return defaultHandler!(ctx);
+    }
+    file = File(resolvedFilePath);
 
-    final entityType = FileSystemEntity.typeSync(fsPath);
+    return await _serveFile(
+      file,
+      mimeResolver ?? _defaultMimeTypeResolver,
+      cacheControl,
+      ctx,
+    );
+  };
+}
 
-    File? fileFound;
+/// Creates a Relic [Handler] that serves a single file at [filePath].
+///
+/// The file must exist at the specified path or an [ArgumentError] is thrown.
+/// Supports the same features as [createStaticHandler] but for a single file.
+///
+/// The [mimeResolver] can be provided to customize MIME type detection.
+/// The [cacheControl] header can be customized; defaults to no-cache with private cache.
+Handler createFileHandler(
+  final String filePath, {
+  final MimeTypeResolver? mimeResolver,
+  final CacheControlHeader? cacheControl,
+}) {
+  final file = File(filePath);
+  if (!file.existsSync()) {
+    throw ArgumentError(
```

Comment: Nice that we fail early on this 🌟